ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java

```diff
@@ -126,9 +127,17 @@
       "ffab", "ffcm", "ffhp", "ffip", "ffit", "ffnt", "ffocus", "ffos", "ffsb", "fpas", "fpsa",
       "ftas", "ftsa", "newext",
       // Yahoo
-      "monline_dg", "monline_3_dg", "monline_4_dg", "monline_7_dg"
-  // End of copied values.
-  );
+      "monline_dg", "monline_3_dg", "monline_4_dg", "monline_7_dg",
+      // End copied desktop codes.
+      // Additionally, these Baidu codes are relevant for mobile browsers.
+      "1000969a", "1000969b");
+
+  private static final Set<String> ALLOWED_SEARCH_CODES_LOWER = ALLOWED_SEARCH_CODES
```

Comment:
- I don't think `ALLOWED_SEARCH_CODES_LOWER` quite accurately captures what's happening, since there's also the leading underscores mobile adds to codes starting with digits (it baffles me why they're doing that in the mobile client).
- If desktop ever adds a search code that starts with a digit I don't believe this arrangement would continue work, because AFAIK desktop isn't adding leading underscores like mobile does.

I'd suggest:
- Renaming `ALLOWED_SEARCH_CODES` to `ALLOWED_SEARCH_CODES_DESKTOP` and removing the mobile-only Baidu codes from that list.
- Renaming this `ALLOWED_SEARCH_CODES_MOBILE` and construct it as `ALLOWED_SEARCH_CODES_DESKTOP` + Baidu codes without leading underscores -> lowercase -> add leading underscores to codes starting with digits.
- Using the those two sets as appropriate in the `sanitizeDesktopSearchKeys` and `sanitizeMobileSearchKeys` methods rather than trying to use a single `isAllowedSearchCode` method.